### PR TITLE
Extract the video key from a YouTube link

### LINKED ISSUE #55: Allow YouTube URL form to take actual youtube URL or youtube key
cursor_and_localhost_5000_dev_html___tournamentid_6

The YouTube URL form actually only works if you paste a youtube key like `y5sdfa834lJ`.  This key is what should be stored in the DB, but we should allow people to paste either the key or the URL and it works.

### PR DESCRIPTION
**Pull request by akagomez**
- Define `serialize` and `get` properties for the Game model's `videoUrl` property
- Create a test environment for the Game model (html/js)
- Test various YouTube urls as `videoUrl`'s

Closes #55.

### CODE REVIEW

**Comment by justinbmeyer** [comment]
I was probably just going to do this w/ a getter /setter

File: public/models/game/game.js [hunk 1]
```diff
@@ -59,6 +59,18 @@ var Game = Map.extend({
 				newVal.__listSet = {gameId: this.attr("id")};
 				return newVal;
 			}
+		},
+		videoUrl: {
+			serialize: true,
+			get: function (setVal) {
```
Comment (justinbmeyer) [inline public/models/game/game.js]: without a setter, how are you testing what is actually sent to the server?

Comment (akagomez) [inline public/models/game/game.js]: I don't fully understand the question. I test the value of the property after the `set` using `.attr('videoUrl`)`. How would the existence of a setter change how I test what's sent to the server?

Comment (justinbmeyer) [inline public/models/game/game.js]: nm ... I forgot that serialize calls `.get` internally.  Still, I would probably solve this with a setter.  That way the stored value is always the right value.

Comment (akagomez) [inline public/models/game/game.js]: Noted.

Comment (justinbmeyer) [inline public/models/game/game.js]: This is a must fix at some point.  So good to change now.  Basically the same logic in `set` is all that's needed.

Comment (akagomez) [inline public/models/game/game.js]: Will do.

**Pull request merged**